### DISCUSSION TITLE
Best AIR IDE?

### DISCUSSION THREAD

**Discussion by slendertech** [airsdk/Adobe-Runtime-Support#3309] [General]
So, in the modern day, I'm not really able to just google the best IDEs to use for creating AIR apps with the Harmon AIR SDK. That's why I'm asking here. Most of the free apps that worked back in the day like FlashDevelop are now very outdated and have not been updated in years. But what about programs like JetBrains IntelliJ? Even though it isn't free, could I use that to create AIR apps with the flash extension?

Any advice would be greatly appreciated.

**Comment by slendertech** [comment]
Okay, sorry. Didn't realize there was an article on this on the website.

**Comment by ylazy** [comment]
I'm using VSCode + the [ActionScript & MXML](https://marketplace.visualstudio.com/items?itemName=bowlerhatllc.vscode-as3mxml) extension + [some essential extensions](https://github.com/ylazy/vscode-actionscript-extension-pack) that can be used with Air.

Once you use VSCode + AS3MXML to develop Air Apps, you won't want to switch to another IDE.

**Comment by marchbold** [comment]
Currently the best IDEs are either IntelliJ IDEA or Visual Studio Code. I use both of those regularly. Some people are using the Moonshine IDE which have heard decent reports about. Animate is still widely used to but I tend only to use this for resource creation. 

Best info is here: https://airsdk.dev/docs/basics/setup-an-editor
